### PR TITLE
Sushi-Scan: Fix images not loading

### LINKED ISSUE #18520: Sushi-Scan: Image could not be loaded
### Source information

Sushi-Scan 1.4.33

### Source language

French

### Steps to reproduce

Opening any manga fails and shows "image could not be loaded"

### Expected behavior

Displays the selected manga

### Actual behavior

Opening any manga fails and shows "image could not be loaded"

### Tachiyomi version

0.14.6

### Android version

Android 12

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.14.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by KirinRaikage**
Checklist:

Closes #18520 

- [ ] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by KirinRaikage** [comment]
@AwkwardPeak7 Should I undo the ts_reader thingy?

**Comment by AwkwardPeak7** [comment]
It's fine to keep these changes as a refactor imo.

**Comment by KirinRaikage** [comment]
Okay, I keep things as is then!

File: multisrc/overrides/mangathemesia/sushiscan/src/SushiScan.kt [hunk 1]
```diff
@@ -25,28 +27,32 @@ class SushiScan : MangaThemesia("Sushi-Scan", "https://sushiscan.net", "fr", man
...
     override fun pageListParse(document: Document): List<Page> {
-        val htmlPages = document.select(pageSelector)
-            .filterNot { it.imgAttr().isEmpty() }
-            .mapIndexed { i, img -> Page(i, "", if (img.imgAttr().contains("https")) img.imgAttr() else img.imgAttr().replace("http", "https")) }
+        val scriptContent = document.selectFirst("script:containsData(ts_reader)")?.data()
+            ?: return super.pageListParse(document)
+        val jsonString = scriptContent.substringAfter("ts_reader.run(").substringBefore(");")
+        val tsReader = json.decodeFromString<TSReader>(jsonString)
+        val imageUrls = tsReader.sources.firstOrNull()?.images ?: return emptyList()
+        return imageUrls.mapIndexed { index, imageUrl -> Page(index, imageUrl = imageUrl) }
```
Comment (AwkwardPeak7) [inline multisrc/overrides/mangathemesia/sushiscan/src/SushiScan.kt:36]:
```suggestion
        return imageUrls.mapIndexed { index, imageUrl -> Page(index, document.location(), imageUrl) }
```

File: multisrc/overrides/mangathemesia/sushiscan/src/SushiScan.kt [hunk 2]
```diff
@@ -25,28 +27,32 @@ class SushiScan : MangaThemesia("Sushi-Scan", "https://sushiscan.net", "fr", man
...
+    }
 
-        countViews(document)
+    override fun imageRequest(page: Page): Request {
```
Comment (AwkwardPeak7) [inline multisrc/overrides/mangathemesia/sushiscan/src/SushiScan.kt:39]: can remove this override with `page.url` populated

Comment (KirinRaikage) [reply multisrc/overrides/mangathemesia/sushiscan/src/SushiScan.kt]: Thanks, done

**Pull request merged**